### PR TITLE
setup lib.rs and bin coexistence

### PR DESCRIPTION
**Pull request by tbmreza**
The main thing I'm trying to do in this pull-request is to add the file replibyte/tests/mysql.rs. In addition to allowing importing modules from tests folder, it also revealed that the current structure allows for 'clever' use statements such as
```rust
// replibyte/src/config.rs
use crate::{RandomTransformer, Transformer};

// or replibyte/src/tasks.rs
use crate::Source;
```
which is saying "use RandomTransformer, Transformer that are imported from crate (main.rs)". If main.rs in the future doesn't use them anymore, replibyte/src/config.rs (and its dependent) would break.

Please let me know what you think 🤓

### CODE REVIEW

File: replibyte/src/source/mysql.rs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+impl<'a> Database for Mysql<'a> {
+    fn stream_rows<F: FnMut(OriginalRow, Row)>(
+        &self,
+        transformers: &Vec<Box<dyn Transformer + '_>>,
+        mut row: F,
+    ) -> Result<(), Error> {
+        let s_port = self.port.to_string();
+
+        let mut process = Command::new("pg_dump")
```
Comment (pepoviola) [inline replibyte/src/source/mysql.rs:61]: This should be `mysqldump` commands.

File: replibyte/src/bin/example.rs [hunk 2]
```diff
@@ -1,27 +1,14 @@
-use crate::bridge::s3::S3;
-use crate::config::{Config, ConnectionUri};
+use replibyte::bridge::s3::S3;
```
Comment (pepoviola) [inline replibyte/src/bin/example.rs:1]: I think maybe `cli.rs` is a better name, at some point we will want to add a cli.

File: replibyte/src/source/mysql.rs [hunk 3]
```diff
@@ -0,0 +1,256 @@
+use std::collections::HashMap;
+use std::io::{BufReader, Error, ErrorKind};
+use std::process::{Command, Stdio};
+
+use dump_parser::postgres::{
```
Comment (pepoviola) [inline replibyte/src/source/mysql.rs:5]: We may want to abstract some of this to have to use from `...::postgres::..`.

Comment (evoxmusic) [reply replibyte/src/source/mysql.rs:5]: agree

**Comment by evoxmusic** [comment]
@tbmreza out of topic but it would be great to also work on a valid Dockerfile to run RepliByte as we need to include `pg_dump` and `mysqldump` binaries into it. We also need to check that those binaries are available in our Postgres and MySQL connectors. WDYT?

**Comment by tbmreza** [comment]
@evoxmusic yeah I think Dockerfile would be convenient. Is that going to be our permanent dependency, or eventually it's going to be all Rust?

In any case, I don't think I know everything that entails to open an issue for it. I am still wrapping my head around Replibyte's end-to-end process.

File: replibyte/src/source/mysql.rs (hunk 3 above)
Comment (tbmreza) [reply replibyte/src/source/mysql.rs:5]: Could you explain more? Is that something I should fix now or for the future? @pepoviola

**Comment by evoxmusic** [comment]
@tbmreza does this PR is ready to be merged for you?

**Comment by tbmreza** [comment]
yes, I'm happy with it for now @evoxmusic

**Comment by evoxmusic** [comment]
I made some structural changes that is breaking your PR. Can you fix it?

**Comment by tbmreza** [comment]
alrighty. give me one sec @evoxmusic

**Comment by tbmreza** [comment]
done @evoxmusic

**Comment by evoxmusic** [comment]
I am not sure this PR is still relevant. Please feel free to re-open it if it is the case. Thank you @tbmreza

**Pull request closed without merging**